server/xpub-server/entities/manuscript/resolvers/index.js

```diff
@@ -28,10 +28,20 @@ const resolvers = {
       return manuscript.save()
     },
 
-    // TODO restrict this in production
     async deleteManuscript(_, { id }, { user }) {
       const userUuid = await User.getUuidForProfile(user)
       const manuscript = await Manuscript.find(id, userUuid)
+
+      const manuscriptFile = manuscript.files[0]
+      if (manuscriptFile) {
+        try {
+          const file = await File.find(manuscriptFile.manuscriptId)
+          await file.deleteContent()
```

Comment:
Architecture ... do we call out to the different areas of concern here (i.e. model /s3) ... or do we introduce another layer (logic) that will handle the orchestrations of permissions, model, s3, etc. ?